pkg/agent/controller/networkpolicy/fqdn.go

```diff
@@ -533,6 +531,13 @@ func (rst *ruleSyncTracker) subscribe(waitCh chan error, dirtyRules sets.Set[str
 	}
 }
 
+// getDirtyRules retrieves the current dirty rule set of ruleSyncTracker.
+func (rst *ruleSyncTracker) getDirtyRules() sets.Set[string] {
+	rst.mutex.Lock()
```

Comment:
I'm wondering if we could just use read lock here and in `f.ruleSyncTracker.subscribe()`?

Reply:
Not for `subscribe`. It updates the `ruleToSubscriber` map which is a write operation